### PR TITLE
Fix Sidebar navigations being cancelled.

### LINKED ISSUE #6585: Links in amp-sidebar stopped working in Chrome
Examples: 

* https://ampbyexample.com
* https://www.webton.be/wie-zijn-wij/?wpamp

Steps to reproduce:

Click on any of the links in the menu in https://ampbyexample.com. The amp-sidebar closes, but doesn't open the link.

Reproduceable in Chrome on Desktop and Android (Version 54.0.2840.98). 
Works in Safari.

This got originally reported here:

https://github.com/ampproject/amp-by-example/issues/521

Update: AMP Version 1481049920709

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes #6585

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -132,10 +133,23 @@ export class AmpSidebar extends AMP.BaseElement {
...
 
+    // TODO(mkhatib, #6589): Consider exposing onLocalNavigation from
+    // document-click service to simplifiy this.
     this.element.addEventListener('click', e => {
       const target = closestByTag(dev().assertElement(e.target), 'A');
       if (target && target.href) {
-        this.close_();
+        const tgtLoc = parseUrl(target.href);
+        const curLoc = parseUrl(this.getAmpDoc().win.location.href);
+        const tgtHref = `${tgtLoc.origin}${tgtLoc.pathname}${tgtLoc.search}`;
```
Comment (cramforce) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Instead compare `removeFragment(a.href)  == removeFragment(b.href)`

Comment (mkhatib) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Done. Should i keep the check for `tgt.hash` still or just close if the navigation is not to a new page? I think it's not entirely needed but a bit safer and we can relax it moving forward.

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js [hunk 2]
```diff
@@ -282,11 +282,91 @@ describe('amp-sidebar', () => {
     });
   });
 
+  it('should reflect state of the sidebar', () => {
```
Comment (mkhatib) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: Not sure why github thinks I added this :\.

Comment (mkhatib) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: I think the git diff just messed up

**Comment by mkhatib** [comment]
Added tests PTAL 👀

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js [hunk 3]
```diff
@@ -340,22 +429,33 @@ describe('amp-sidebar', () => {
...
+        return {
+          win: {
+            location: {
+              // Mocking navigating from example.com -> localhost:9876
+              href: 'http://example.com',
+            },
+          },
+        };
+      });
+      anchor.dispatchEvent ?
```
Comment (camelburrito) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: not blocking you on this , may be we should move this to a integration test.

Comment (mkhatib) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: Filed #6593 to refactor this in future

**Pull request merged**